### PR TITLE
Inflight I/O: Make struct members public

### PR DESCRIPTION
**Pull request by harshanavkis**
Signed-off-by: Harshavardhan Unnibhavi <harshanavkis@gmail.com>

### CODE REVIEW

**Comment by harshanavkis** [comment]
@sboeuf for the inflight I/O implementation, for now, desc_num field needs to be updated. Hence I made all fields public. Would this be alright or do you prefer to write setter methods for each field individually.

File: src/vhost_user/message.rs [hunk 1]
```diff
@@ -786,13 +786,13 @@ impl VhostUserMsgValidator for VhostUserFSSlaveMsg {
 #[derive(Clone, Copy, Default)]
 pub struct DescStateSplit {
     /// Indicate whether this descriptor (only head) is inflight or not.
-    inflight: u8,
+    pub inflight: u8,
     /// Padding
-    padding: [u8; 5],
+    pub padding: [u8; 5],
```
Comment (jiangliu) [inline src/vhost_user/message.rs:791]: Do we need to export padding too?

**Review by sboeuf — COMMENTED** [review]
PR sounds good to me, but I agree with @jiangliu that you shouldn't make the `padding` fields public.

File: src/vhost_user/message.rs [hunk 2]
```diff
@@ -839,25 +839,25 @@ impl QueueRegionSplit {
 #[derive(Clone, Copy, Default)]
 pub struct DescStatePacked {
     /// Indicate whether this descriptor (only head) is inflight or not.
-    inflight: u8,
+    pub inflight: u8,
     /// Padding
-    padding: u8,
+    pub padding: u8,
```
Comment (sboeuf) [inline src/vhost_user/message.rs:844]: Ditto about `padding`.

File: src/vhost_user/message.rs [hunk 3]
```diff
@@ -872,27 +872,27 @@ impl DescStatePacked {
...
+    pub old_used_idx: u16,
     /// device ring wrap counter
-    used_wrap_counter: u8,
+    pub used_wrap_counter: u8,
     /// old device ring wrap counter
-    old_used_wrap_counter: u8,
+    pub old_used_wrap_counter: u8,
     /// Padding
-    padding: [u8; 7],
+    pub padding: [u8; 7],
```
Comment (sboeuf) [inline src/vhost_user/message.rs:893]: Ditto about `padding`.

**Comment by harshanavkis** [comment]
@jiangliu @sboeuf makes sense to me. I updated the PR.

**Review by sboeuf — APPROVED** [review]

File: src/vhost_user/message.rs [hunk 4]
```diff
@@ -872,27 +872,27 @@ impl DescStatePacked {
 #[repr(packed)]
```
Comment (jiangliu) [inline src/vhost_user/message.rs:872]: Why `repr(packed)` instead of `repr(c)` here and above?

**Review by jiangliu — APPROVED** [review]

**Pull request merged**